### PR TITLE
Crop Cover Patterns

### PR DESCRIPTION
**Pull request by rajadain**
## Overview

To differentiate cover crops from regular crops, we add an overlay pattern to them. In addition, we add a new script to automate and maintain consistency of colors across the app. This script also has variables to easily change pattern parameters.

Connects #154 

### Demo

image

## Testing Instructions

 * Check out the branch, ensure the colors match up in general (across charts, thumbnails, shapes on map)
 * Ensure cover crops have a cross-hatch pattern on them
 * Run `./scripts/colors/update.py` and ensure that no files are changed in git (new files should be identical to old ones)
 * Try changing the stroke width, or line color, or a color in `cdl_colormap`, and run the script again. Ensure the changes carry through. Run bundle, and ensure new colors / patterns are visible in the app.

### CODE REVIEW

File: scripts/colors/update.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+    patterns.append('''
+        <pattern id="fill-crop-{id}" patternUnits="userSpaceOnUse" width="{size}" height="{size}">
+            <rect width="{size}" height="{size}" style="fill:{color}" />{lines}
+        </pattern>
+    '''.format(id=key, size=PATTERN_SIZE, color=color, lines=lines))
+
+    styles.append('{}: {}'.format(key, color))
+
+target = open(PATTERNS_FILE, 'w')
```
Comment (mmcfarland) [inline scripts/colors/update.py]: It's more idiomatic to do file io in python using the `with` statement to ensure the resources are closed despite exceptions, etc.

``` python
with open(PATTERNS_FILE, 'w') as target:
    target.write('...etc...')
```

Consider that technique here and below.

**Comment by rajadain** [comment]
Linted the file, added it to lint checking, and attempted a hexagon pattern that almost worked:

image

**Comment by rajadain** [comment]
Improved:

image

**Comment by rajadain** [comment]
Thanks, will do. That's why I kept it separate.

**Pull request merged**